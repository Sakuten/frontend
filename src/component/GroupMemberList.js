import React from 'react'
import styled from 'styled-components'
import { observer } from 'mobx-react'

const PublicIDWrap = styled.span`
  width: 3rem;
  display: inline-block;
`

const Container = styled.div`
  margin: 5px;
`

const GroupMemberList = ({list, onRemove}) => (
  <div>
    {
      list.length !== 0 ? list.map((publicId, i) =>
        <Container data-test='groupmemberlist-member' key={i}>
          <PublicIDWrap>
            {publicId}
          </PublicIDWrap>
          <button className='button is-danger is-outlined' data-test='groupmemberlist-remove' onClick={() => onRemove(i)}>削除</button>
        </Container>
      ) : <span data-test='groupmemberlist-notfound'>一緒に応募する人はいません。</span>
    }
  </div>
)

export default observer(GroupMemberList)
