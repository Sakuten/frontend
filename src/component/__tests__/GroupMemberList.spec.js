import React from 'react'
import { shallow } from 'enzyme'
import GroupMemberList from '../GroupMemberList'

const setup = propOverrides => {
  const props = Object.assign({
    list: [
      'public_id1',
      'public_id2',
      'public_id3'
    ],
    onRemove: jest.fn()
  }, propOverrides)

  const wrapper = shallow(<GroupMemberList {...props} />)

  return {
    props,
    wrapper,
    member: wrapper.find('[data-test="groupmemberlist-member"]'),
    removeButton: wrapper.find('[data-test="groupmemberlist-remove"]'),
    notfound: wrapper.find('[data-test="groupmemberlist-notfound"]')
  }
}

describe('components', () => {
  describe('GroupMemberList', () => {
    it('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders a message when the list was empty', () => {
      const { notfound } = setup({list: []})
      expect(notfound.length).toBe(1)
    })

    it('renders two members', () => {
      const { member } = setup()
      expect(member.length).toBe(3)
    })

    it('renders two cancel buttons', () => {
      const { removeButton } = setup()
      expect(removeButton.length).toBe(3)
    })

    it('calls onRemove with index when remove button is clicked', () => {
      const mock = jest.fn()
      const { removeButton } = setup({onRemove: mock})
      removeButton.at(2).simulate('click')
      expect(mock).toBeCalledWith(2)
    })
  })
})
