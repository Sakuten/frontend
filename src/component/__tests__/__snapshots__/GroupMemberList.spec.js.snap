// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components GroupMemberList render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupMemberList
    list={
      Array [
        "secret_id1",
        "secret_id2",
        "secret_id3",
      ]
    }
    onRemove={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          data-test="groupmemberlist-member"
        >
          secret_id1
          <button
            data-test="groupmemberlist-remove"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>,
        <div
          data-test="groupmemberlist-member"
        >
          secret_id2
          <button
            data-test="groupmemberlist-remove"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>,
        <div
          data-test="groupmemberlist-member"
        >
          secret_id3
          <button
            data-test="groupmemberlist-remove"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "secret_id1",
            <button
              data-test="groupmemberlist-remove"
              onClick={[Function]}
            >
              Remove
            </button>,
          ],
          "data-test": "groupmemberlist-member",
        },
        "ref": null,
        "rendered": Array [
          "secret_id1",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Remove",
              "data-test": "groupmemberlist-remove",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Remove",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "secret_id2",
            <button
              data-test="groupmemberlist-remove"
              onClick={[Function]}
            >
              Remove
            </button>,
          ],
          "data-test": "groupmemberlist-member",
        },
        "ref": null,
        "rendered": Array [
          "secret_id2",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Remove",
              "data-test": "groupmemberlist-remove",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Remove",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "secret_id3",
            <button
              data-test="groupmemberlist-remove"
              onClick={[Function]}
            >
              Remove
            </button>,
          ],
          "data-test": "groupmemberlist-member",
        },
        "ref": null,
        "rendered": Array [
          "secret_id3",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Remove",
              "data-test": "groupmemberlist-remove",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Remove",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            data-test="groupmemberlist-member"
          >
            secret_id1
            <button
              data-test="groupmemberlist-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>,
          <div
            data-test="groupmemberlist-member"
          >
            secret_id2
            <button
              data-test="groupmemberlist-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>,
          <div
            data-test="groupmemberlist-member"
          >
            secret_id3
            <button
              data-test="groupmemberlist-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "secret_id1",
              <button
                data-test="groupmemberlist-remove"
                onClick={[Function]}
              >
                Remove
              </button>,
            ],
            "data-test": "groupmemberlist-member",
          },
          "ref": null,
          "rendered": Array [
            "secret_id1",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Remove",
                "data-test": "groupmemberlist-remove",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Remove",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "secret_id2",
              <button
                data-test="groupmemberlist-remove"
                onClick={[Function]}
              >
                Remove
              </button>,
            ],
            "data-test": "groupmemberlist-member",
          },
          "ref": null,
          "rendered": Array [
            "secret_id2",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Remove",
                "data-test": "groupmemberlist-remove",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Remove",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "secret_id3",
              <button
                data-test="groupmemberlist-remove"
                onClick={[Function]}
              >
                Remove
              </button>,
            ],
            "data-test": "groupmemberlist-member",
          },
          "ref": null,
          "rendered": Array [
            "secret_id3",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Remove",
                "data-test": "groupmemberlist-remove",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Remove",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
